libraries/AP_InertialNav/AP_InertialNav_NavEKF.cpp

```diff
@@ -53,11 +53,8 @@ nav_filter_status AP_InertialNav_NavEKF::get_filter_status() const
  */
 Location AP_InertialNav_NavEKF::get_origin() const
 {
-    Location ret;
-     if (!_ahrs_ekf.get_origin(ret)) {
-         // initialise location to all zeros if EKF origin not yet set
-         memset(&ret, 0, sizeof(ret));
-     }
+    Location ret;  // Location is initialised to all zeros in case EKF origin not set
+    _ahrs_ekf.get_origin(ret);
```

Comment:
Currently this is valid, but we've been known to fill in these structs to the best of our ability before, and then return false. If we ever changed the origin to behave like that this change would break.

I also don't like it because we should be tagging methods like `AP_AHRS::get_origin(Location &ret)` with `WARN_IF_UNUSED` which except for one bug in `AC_Fence` actually already works... In fact I went off and PR'd it: #10783

Reply:
I guess I could keep the check and use ret.zero() instead of memset to keep gcc happy. The behavior will be the same

Reply:
Yeah, I'd be happier with that as it's less of a future trap (and also allows the `WARN_IF_UNUSED` thing to continue :) )

Reply:
yup ! Not a problem !